aws/resource_aws_codebuild_project.go

```diff
@@ -659,6 +679,22 @@ func expandProjectEnvironment(d *schema.ResourceData) *codebuild.ProjectEnvironm
 		projectEnv.ImagePullCredentialsType = aws.String(v.(string))
 	}
 
+	if v := envConfig["registry_credential"]; v != nil {
+		config := v.(*schema.Set).List()[0].(map[string]interface{})
+
+		projectRegistryCredential := &codebuild.RegistryCredential{}
+
+		if v := config["credential"].(string); v != "" {
+			projectRegistryCredential.Credential = &v
```

Comment:
Generally when setting AWS Go SDK parameters, we prefer (by convention) to use the provided [conversion functions](https://docs.aws.amazon.com/sdk-for-go/api/aws/) (e.g. `aws.String()`) instead of raw Go pointer referencing

```suggestion
			projectRegistryCredential.Credential = aws.String(v.(string))
```